plugins/source/azure/client/client.go

```diff
@@ -54,20 +56,25 @@ func (c *Client) discoverSubscriptions(ctx context.Context) error {
 	return nil
 }
 
+func getResourceGroupNames(resourceGroups []*armresources.ResourceGroup) []string {
+	names := make([]string, len(resourceGroups))
+	for i, rg := range resourceGroups {
+		names[i] = *rg.Name
+	}
+	return names
+}
+
 func (c *Client) disocverResourceGroups(ctx context.Context) error {
-	c.resourceGroups = make(map[string][]*armresources.GenericResourceExpanded, len(c.subscriptions))
-	filter := "resourceType eq 'Microsoft.Resources/resourceGroups'"
```

Comment:
This query wasn't working (returned 0 results). There's a dedicated client to get the resource groups so I used that one